File: source/fortran/CS_fatigue_variables.f90
module CS_fatigue_variables

#ifndef dp
  use, intrinsic :: iso_fortran_env, only: dp=>real64
#endif

  implicit none

  public

  real(dp) :: residual_sig_hoop
  !! residual hoop stress in strucutal material (Pa)

  real(dp) :: n_cycle
  !! Allowable number of cycles for CS stress model

  real(dp) :: n_cycle_min
  !! Minimum llowable number of cycles for CS stress model

  real(dp) :: t_crack_radial
  !! Initial depth of crack in thickness of conduit (m)

  real(dp) :: t_crack_vertical
  !! Inital vertical crack size (m)

  real(dp) :: t_structural_radial
  !! Thickness of CS conductor conduit (m)

  real(dp) :: t_structural_vertical
  !! Vertical thickness of CS conductor conduit (m)

  real(dp) :: bkt_life_csf
  !! Switch to pass bkt_life cycles to n_cycle_min

  real(dp) :: sf_vertical_crack
  !! Safety factor for vertical crack size (-)

  real(dp) :: sf_radial_crack
  !! Safety factor for radial crack size (-)

  real(dp) :: sf_fast_fracture
  !! safety factor for stress intensity factor (-)

  real(dp) :: paris_coefficient
  !! Paris equation material coefficient (-)

  real(dp) :: paris_power_law
  !! Paris equation material power law (-)

  real(dp) :: walker_coefficient
  !! walker coefficent (-)

  real(dp) :: fracture_toughness
  !! fracture toughness (MPa m^1/2)
end module CS_fatigue_variables

File: process/data_structure/cs_fatigue_variables.py
"""Module containing variables for the CS fatigue models"""

from dataclasses import dataclass


@dataclass(slots=True)
class CSFatigueData:
    """Dataclass holding CS fatigue variables"""

    stress_hoop_cs_residual: float = 2.4e8
    """Residual hoop stress in CS structural material (Pa)"""

    n_cycle: float = 0.0
    """Allowable number of cycles for CS stress model"""

    n_cycle_min: float = 2.0e4
    """Minimum allowable number of cycles for CS stress model"""

    dr_cs_turn_crack_initial: float = 6.0e-3
    """Initial CS turn radial crack size thickness [m]"""

    dz_cs_turn_crack_initial: float = 0.89e-3
    """Initial CS turn vertical crack size thickness [m]"""

    dr_cs_turn_conduit: float = 0.07
    """Thickness of CS conductor conduit (m)"""

    dz_cs_turn_conduit: float = 0.022
    """Vertical thickness of CS conductor conduit (m)"""

    bkt_life_csf: float = 0.0
    """Switch to pass bkt_life cycles to n_cycle_min"""

    sf_vertical_crack: float = 2.0
    """Safety factor for vertical crack size (-)"""

    sf_radial_crack: float = 2.0
    """Safety factor for radial crack size (-)"""

    sf_fast_fracture: float = 1.5
    """safety factor for stress intensity factor (-)"""

    paris_coefficient_cs_turn: float = 65.0e-14
    """Paris equation material coefficient / fatigue crack growth coefficient (C)"""

    paris_exponent_cs_turn: float = 3.5
    """Paris equation material power law (-)"""

    walker_coefficient: float = 0.436
    """walker coefficient (-)"""

    fracture_toughness: float = 2.0e2
    """fracture toughness (MPa m^1/2)"""


CREATE_DICTS_FROM_DATACLASS = CSFatigueData
